src/macho.rs

```diff
@@ -1,10 +1,16 @@
+use crate::error::{GetSymbolsError, Result};
 use compact_symbol_table::CompactSymbolTable;
 use object::{MachOFile, Object};
-
-pub fn get_compact_symbol_table(buffer: &[u8], breakpad_id: &str) -> Option<CompactSymbolTable> {
-    let macho_file = MachOFile::parse(buffer).ok()?;
-    if format!("{:X}0", macho_file.mach_uuid()?.simple()) != breakpad_id {
-        return None;
+pub fn get_compact_symbol_table(buffer: &[u8], breakpad_id: &str) -> Result<CompactSymbolTable> {
+    let macho_file = MachOFile::parse(buffer).or(Err(GetSymbolsError::InvalidInputError(
+        "Could not parse Mach-O header",
+    )))?;
+    let macho_id = format!("{:X}0", macho_file.mach_uuid().unwrap().simple());
```

Comment:
There's another unwrap call here.

Reply:
Similarly, we'd have the error ``` the trait `std::convert::From<std::option::NoneError>` is not implemented for `error::GetSymbolsError` ``` if we use `?` instead of `.unwrap()`

Reply:
Ok, then we should handle the None case manually.

```rust
let macho_id = match macho_file.mach_uuid() {
  Some(id) => format!("{:X}0", id.simple());,
  None => return Err(GetSymbolsError::InvalidInputError("Could not get mach uuid"))
};
```
or something similar.